core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -490,11 +490,14 @@ class LogSegment private[log] (val log: FileRecords,
    * Change the suffix for the index and log files for this log segment
    * IOException from this method should be handled by the caller
    */
-  def changeFileSuffixes(oldSuffix: String, newSuffix: String): Unit = {
+  def changeFileSuffixes(oldSuffix: String, newSuffix: String, needFlushParentDir: Boolean = true): Unit = {
```

Comment:
Hmm, we need to pass needFlushParentDir to each of log.renameTo and index.renameTo to disable flushing, right?

Reply:
set the flag to false for rename.